xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -564,8 +564,10 @@ bool CEdl::ReadPvr(const CFileItem &fileItem)
   return !cutlist.empty();
 }
 
-bool CEdl::AddCut(Cut& cut)
+bool CEdl::AddCut(const Cut& NewCut)
```

Comment:
better pass by value instead of making a copy at the begin of the function

Reply:
I don't think this would be better. Separate interface from implementation...

Reply:
How does this relate to `separate interface from implementation`?
I'm more with rechi here, passing by value makes it obvious to the caller that it will be copied, without needing to even read into the implementation.
But just stating my opinion, decision is yours ofc.

Reply:
I see it more like this: Normal way of doing things is to pass "large" objects by reference. If I pass by value there must be good reason for this. The fact that the current implementation wants to manipulate the parameter's value is 100% implementation detail, has nothing to do with the interface and this does not justify to break the rule to pass by value only in exceptional cases. 

Let's assume I change the implementation not to manipulate the parameter's value anymore. I bet, I easily forget to change the parameter to "const &", which I should do for best interface design, because by-value is no longer needed, then. Ooops. Implementation detail controls the interface? Really? Should not be.

Reply:
It's like the difference between `ModifyAndAddCut(Cut& cut)` and `AddCut(const Cut& newCut)`.